common/src/WooCommerce.php

```diff
@@ -60,8 +61,16 @@ public function autoSyncNotice() {
     }
 
     public function set_order_language($order_id) {
-        if (!get_post_meta($order_id, 'wpml_language') && defined('ICL_LANGUAGE_CODE')) {
-            update_post_meta($order_id, 'wpml_language', ICL_LANGUAGE_CODE);
+        if ( class_exists( 'Automattic\WooCommerce\Utilities\OrderUtil' ) && OrderUtil::custom_orders_table_usage_is_enabled() ) {
+            /** @var WC_Order $order */
+            $order = wc_get_order($order_id);
+            if (!$order->get_meta('wpml_language') && defined('ICL_LANGUAGE_CODE')) {
+                $order->update_meta_data('wpml_language', ICL_LANGUAGE_CODE);
+            }
```

Comment:
This technique should always work, independently of whether the custom orders table is used. So I think that the `get_post_meta`/`update_post_meta` code can be dropped.

Reply:
As far as I read here:
https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book
these should be changed, or have I misunderstood it?

Reply:
My point is that we don't need both. We only need the new API, the old API calls can be dropped.

In other words, I believe we can use the new API calls instead of the old API calls, not in addition to.

Reply:
I believe @apeschar meant that you can use `get_meta` and `update_meta_data` without checking if `custom_orders_table_usage_is_enabled`

---

Also in the https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book
It says that `update_meta_data` should be followed by `order->save()`